src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/HandlerMethodDescriptor.cs

```diff
@@ -12,5 +14,7 @@ public class HandlerMethodDescriptor
         public MethodInfo Method { get; set; }
 
         public Func<Page, object, Task<IActionResult>> Executor { get; set; }
+
+        public IEnumerable<IFilterMetadata> Filters {get; set;}
```

Comment:
You want to use these filters to be run in the filter pipeline specifically after a handler is selected. Right now, you're gathering filters on any handler and applying it to all the handlers on the type by collating them. Unless that's specifically the design we're going for (@rynowak?), this isn't correct.